src/adapters/RoomsJSONAdapter.js

```diff
@@ -0,0 +1,83 @@
+import {Observable} from 'rxjs';
+
+import RoomsAdapter from './RoomsAdapter';
+
+/**
+ * @typedef RoomJSON
+ * @param {object} datasource An object that contains a set of rooms keyed by ID.
+ * @example
+ * {
+ *  "room-1": {
+ *     "ID": "room-1",
+ *     "title": "title",
+ *     "roomType": "group"
+ *  }
+ * }
+ */
+
+/*
+ * Implements the RoomsAdapter interface with a JSON object as its datasource. See @RoomsJSON
+ */
+export default class RoomsJSONAdapter extends RoomsAdapter {
+  constructor(datasource) {
+    super(datasource);
+    this.getRoom = this.getRoom.bind(this);
+  }
+
+  /**
+   * Returns an observable that emits room data of the given ID.
+   *
+   * @param {string} ID ID of room to get
+   * @returns {Observable.<Room>}
+   * @memberof RoomsJSONAdapter
+   */
+  getRoom(ID) {
+    return Observable.create((observer) => {
+      if (this.datasource[ID]) {
+        observer.next(this.datasource[ID]);
+      } else {
+        observer.error(new Error(`Could not find room with ID "${ID}"`));
+      }
+
+      observer.complete();
+    });
+  }
+
+  /**
+   * Returns an observable that emits an array of previous activity IDs of the given roomID.
+   *
+   * @param {string} ID  ID of the room to get.
+   * @returns {Observable.<Array.<string>>}
+   * @memberof RoomsAdapter
+   */
+  getPreviousRoomActivities(ID) {
+    return Observable.create((observer) => {
+      if (this.datasource[ID]) {
+        observer.next(this.datasource[ID]);
+      } else {
+        observer.error(new Error(`Could not find previous activities associated to the room with ID "${ID}"`));
+      }
+
+      observer.complete();
+    });
+  }
+
+  /**
+   * Returns an observable that emits an array of current activity IDs of the given roomID.
+   *
+   * @param {string} ID  ID of the room to get.
+   * @returns {Observable.<Array.<string>>}
+   * @memberof RoomsAdapter
+   */
+  getRoomActivities(ID) {
+    return Observable.create((observer) => {
+      if (this.datasource[ID]) {
+        observer.next(this.datasource[ID]);
+      } else {
+        observer.error(new Error(`Could not find current activities associated to the room with ID "${ID}"`));
```

Comment:
As with `getPreviousActivities`, we should send an empty array instead of an error, and potentially log the error to console `Could not retrieve new activity for room "${ID}"`.